staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -216,6 +222,18 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 		} else {
 			requestInfo.Verb = "list"
 		}
+
+		// FIXME: For list and watch requests check if there is field selector matching single object.
+		if values := req.URL.Query()["fieldSelector"]; len(values) > 0 {
+			selector, err := fields.ParseSelector(values[0])
```

Comment:
Apparently I was wrong - hasName is computed based on requestInfo:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/namer.go#L73

and that will be set in this PR. So this will work. Changing.